saleor/core/weight.py

```diff
@@ -70,16 +70,16 @@ def format_value(self, value):
             return value.value
         return value
 
-    def render(self, name, value, attrs=None):
-        widget = super().render(name, value, attrs=attrs)
+    def render(self, name, value, attrs=None, renderer=None):
+        widget = super().render(name, value, attrs=attrs, renderer=renderer)
         unit = get_default_weight_unit()
         translated_unit = dict(WeightUnits.CHOICES)[unit]
         return render_to_string(
             self.template,
             {'widget': widget, 'value': value, 'unit': translated_unit})
 
 
-class WeightField(forms.DecimalField):
+class WeightField(forms.FloatField):
```

Comment:
@Pacu2 could you take a look at this? To tell if we keep it that way or refactor the validation. I have an issue with `measurement.measures.Weight` converting the given double value to a float.
```python
    def to_python(self, value):
        value = super().to_python(value)
        if value is None:
            return value
        unit = get_default_weight_unit()
        return Weight(**{unit: value})
```

Because `DecimalField.validate` is expecting a double value, not a float.

Reply:
I think we can keep it this way, it's more reasonable than working around the `Weight` instance (which already has its issues...)